test/unit/generator/content/part2/map-reductions.test.js

```diff
@@ -0,0 +1,88 @@
+    expect(result[0].id).toBe(1)
+  })
+
+  test('maps reason to reduction if one reduction', () => {
+    const result = mapReductions([{ reductions: [{ reason: 'Late claim submission' }] }])
+    expect(result[0].reason).toBe('Late claim submission')
+  })
+
+  test('returns reductions if multiple reductions', () => {
+    const result = mapReductions([{ reductions: [{ reason: 'Late claim submission' }] }, { reductions: [{ reason: 'Over declaration reduction' }] }])
+    expect(result.length).toBe(2)
+  })
+
+  test('maps id to reduction if multiple reductions', () => {
+    const result = mapReductions([{ reductions: [{ reason: 'Late claim submission' }] }, { reductions: [{ reason: 'Over declaration reduction' }] }])
+    expect(result[0].id).toBe(1)
+    expect(result[1].id).toBe(2)
+  })
+
+  test('maps reason to reduction if multiple reductions', () => {
+    const result = mapReductions([{ reductions: [{ reason: 'Late claim submission' }] }, { reductions: [{ reason: 'Over declaration reduction' }] }])
+    expect(result[0].reason).toBe('Late claim submission')
+    expect(result[1].reason).toBe('Over declaration reduction')
+  })
+
+  test('maps reason to reduction if duplicate reductions in multiple fundings', () => {
+    const result = mapReductions([{ reductions: [{ reason: 'Late claim submission' }] }, { reductions: [{ reason: 'Late claim submission' }] }])
+    expect(result[0].reason).toBe('Late claim submission')
+    expect(result.length).toBe(1)
+  })
+
+  test('maps reason to reduction if duplicate reductions in same funding', () => {
+    const result = mapReductions([{ reductions: [{ reason: 'Late claim submission' }, { reason: 'Late claim submission' }] }])
+    expect(result[0].reason).toBe('Late claim submission')
+    expect(result.length).toBe(1)
+  })
+
+  test('maps reason to reduction if duplicate reductions in multiple fundings with non-duplicates', () => {
+    const result = mapReductions([
+      { reductions: [{ reason: 'Late claim submission' }, { reason: 'Over declaration reduction' }] },
+      { reductions: [{ reason: 'Late claim submission' }, { reason: 'Late amendment reduction' }] }
+    ])
+    expect(result[0].reason).toBe('Late claim submission')
+    expect(result[1].reason).toBe('Over declaration reduction')
+    expect(result[2].reason).toBe('Late amendment reduction')
+    expect(result.length).toBe(3)
+  })
+
+  test('maps reason to reduction if duplicate reductions in same fundings with non-duplicates', () => {
+    const result = mapReductions([{
+      reductions: [
+        { reason: 'Late claim submission' },
+        { reason: 'Over declaration reduction' },
+        { reason: 'Late claim submission' },
+        { reason: 'Late amendment reduction' }
+      ]
+    }])
+    expect(result[0].reason).toBe('Late claim submission')
+    expect(result[1].reason).toBe('Over declaration reduction')
+    expect(result[2].reason).toBe('Late amendment reduction')
+    expect(result.length).toBe(3)
```

Comment:
nice tests here -- could you split into single/more granular assertions:

- `length`
- each `id`
- each `reason`

same as you did for lines 34 - 49

Reply:
I've split by reasons and length as suggested.  I did try each category but  felt that it was harder for a reader to interpret the expected outcome to the input without knowing which other tests that fit the collection.